front-end/e-learning/src/components/categories/Create.js

```diff
@@ -0,0 +1,184 @@
+  const [openNotification, setOpenNotification] = useState(false);
+
+  const handleOpen = () => setOpen(true);
+  const handleClose = () => setOpen(false);
+
+  const onCreate = e => {
+    e.preventDefault();
+    const data = {
+      title: title,
+      description: description,
+    };
+    createCategory(data, token);
+    setSubmit(true);
+  };
+
+  const Alert = React.forwardRef(function Alert(props, ref) {
+    return <MuiAlert elevation={6} ref={ref} variant="filled" {...props} />;
+  });
+
+  const handleNotificationClose = () => {
+    setOpenNotification(false);
+    setSubmit(false);
+  };
+
+  useEffect(() => {
+    if (categories.create_error === undefined) setOpen(false);
+    else if (isNull(categories.create_error)) {
+      setOpen(false);
+      setTitle("");
+      setDescription("");
+      setOpenNotification(true);
+    } else setOpen(true);
+  }, [categories]);
+
+  return (
+    <div className='container'>
+      <Button
+        onClick={handleOpen}
+        sx={{
+          mt: 2,
+          mb: 2,
+          bgcolor: "#05386b",
+          color: "white",
+          "&:hover": {
+            backgroundColor: "#edf5e1",
+            color: "black",
+          },
+        }}
+      >
+        Create Category
+      </Button>
+      <Modal open={open} onClose={handleClose}>
+        <Card
+          sx={{
+            width: "60%",
+            position: "absolute",
+            left: "50%",
+            top: "50%",
+            transform: "translate(-50%, -50%)",
+            display: "flex",
+            flexDirection: "column",
+            bgcolor: "white",
+            p: 2,
+            boxShadow: 12,
+          }}
+        >
+          <CardHeader title={"Add Category"} />
+          <CardContent>
+            <form>
+              <TextField
+                margin='normal'
+                required
+                fullWidth
+                id='title'
+                label='Title'
+                name='title'
+                autoComplete='title'
+                autoFocus
+                onChange={e => setTitle(e.target.value)}
+              />
+              <TextField
+                margin='normal'
+                required
+                fullWidth
+                multiline
+                rows={4}
+                name='description'
+                label='Description'
+                type='description'
+                id='description'
+                autoComplete='description'
+                onChange={e => setDescription(e.target.value)}
+              />
+              <Box
+                sx={{
+                  display: "flex",
+                  alignItems: "center",
+                  flexDirection: "column",
+                  color: "red",
+                }}
+              >
+                {categories.create_error === undefined ||
+                categories.create_error === null
+                  ? ""
+                  : "Fields are required"}
```

Comment:
Instead of doing this, you can add validation in MUI elements. See reference attached. Apply for each `TextField`.
https://mui.com/components/text-fields/#validation

Reply:
So is it fine ma'am not to put validation anymore validation in the backend?

Reply:
But this is noted ma'am. Thank you.

Reply:
@rdelrosariojr its better to have validations in both FE and BE for higher security.